sdcm/nemesis.py

```diff
@@ -37,6 +38,37 @@
 from sdcm.utils import remote_get_file
 
 
+def not_run_on_spot(disrupt_method):
```

Comment:
More elegant solution is to create new class that inherits from Nemesis and override `set_target_node` method:

class NoSpotNemesis(Nemesis)
    def set_target_node(self):
        ...
        the logic here
        ....
class RestartThenRepairNodeMonkey(NoSpotNemesis):
    ...

Reply:
Thought about it. But don't think that inheritance will not be clear.